src/libsystemd-network/sd-dhcp-lease.c

```
@@ -505,6 +505,57 @@ static int lease_parse_domain(const uint8_t *option, size_t len, char **ret) {
         return 0;
 }
 
+static int lease_parse_fqdn(const uint8_t *option, size_t len, char **ret) {
+        _cleanup_free_ char *name = NULL, *normalized = NULL;
+        size_t data_len = len - 3;
+        int r;
+        const uint8_t *data;
+
+        assert(option);
+        assert(ret);
+
+        /* RFC 4702 Section 2
+         *
+         * Byte 0: Flags (S, O, E, N)
+         * Byte 1: RCODE1 (ignored on receipt)
+         * Byte 2: RCODE2 (ignored on receipt)
+         * Bytes 3+: Domain Name */
+
+        if (len <= 3)
+                return 0;
```

Please return -EBADMSG or so. The caller anyway ignores the error.